inbox/events/microsoft_graph_client.py

```diff
@@ -0,0 +1,400 @@
+
+        Returns:
+            Iterable of calendars.
+            https://learn.microsoft.com/en-us/graph/api/resources/calendar
+        """
+        # TODO: Figure out the top limit we can use on this endpoint
+        yield from self._iter("/me/calendars")
+
+    def get_calendar(self, calendar_id: str) -> Dict[str, Any]:
+        """
+        Get a single calendar.
+
+        https://learn.microsoft.com/en-us/graph/api/calendar-get
+
+        Arguments:
+            calendar_id: The calendar id
+
+        Returns:
+            The calendar.
+            https://learn.microsoft.com/en-us/graph/api/resources/calendar
+        """
+        return self.request("GET", f"/me/calendars/{calendar_id}")
+
+    def iter_events(
+        self, calendar_id: str, *, modified_after: Optional[datetime.datetime] = None
+    ) -> Iterable[Dict[str, Any]]:
+        """
+        Lazily iterate events in a calendar.
+
+        https://learn.microsoft.com/en-us/graph/api/user-list-events
+
+        Arguments:
+            calendar_id: The calendar id
+            modified_after: If specified only return events given datetime
+
+        Returns:
+            Iterable of events.
+            https://learn.microsoft.com/en-us/graph/api/resources/event
+        """
+        if modified_after:
+            assert modified_after.tzinfo == pytz.UTC
+            params = {
+                "$filter": f"lastModifiedDateTime gt {modified_after.replace(tzinfo=None).isoformat()}Z"
+            }
+        else:
+            params = None
+        # TODO: Figure out the top limit we can use on this endpoint
+        yield from self._iter(f"/me/calendars/{calendar_id}/events", params=params)
+
+    def get_event(self, event_id: str) -> Dict[str, Any]:
+        """
+        Get a single event.
+
+        https://learn.microsoft.com/en-us/graph/api/event-get
+
+        Arguments:
+            event_id: The event id
+
+        Returns:
+            The event.
+            https://learn.microsoft.com/en-us/graph/api/resources/event
+        """
+        return self.request("GET", f"/me/events/{event_id}")
+
+    def iter_event_instances(
+        self, event_id: str, *, start: datetime.datetime, end: datetime.datetime
+    ) -> Iterable[Dict[str, Any]]:
+        """
+        Lazily expand series master instances.
+
+        https://learn.microsoft.com/en-us/graph/api/event-list-instances
+
+        Arguments:
+            event_id: The event id
+            start: the start of the period
+            end: the end of the period
+
+        Returns:
+            Iterable of event instances.
+            https://learn.microsoft.com/en-us/graph/api/resources/event
+        """
+        assert start.tzinfo == pytz.UTC
+        assert end.tzinfo == pytz.UTC
+        assert end >= start
+        yield from self._iter(
+            f"/me/events/{event_id}/instances",
+            params={
+                "startDateTime": start.isoformat(),
```

Comment:
Good feedback @cschams. When testing I found out that some of the parameters are more lenient, for example `startDateTime` and `endDateTime` will happily accept event dates without time and will just assume that it's midnight UTC. Other places wanted the `Z` with time e.g. `2022-10-17T15:46:36.335288Z` and you would get error otherwise. I just tested that all the places accept the format with `Z` so let's use this everywhere.

Changed in [`7792231` (#371)](https://github.com/closeio/sync-engine/pull/371/commits/7792231a8aed151022e206c87d540bb4b67de927)